### PR TITLE
Assign a new display column when deleting field

### PR DESCRIPTION
**Pull request by melohagan**
## Description
Previously an error toast would display when trying to delete the display column. This was annoying because you would have to untick the display column, save, then edit again to delete.

Now you can delete display columns and a new display column will automatically be chosen.

I also fixed a minor bug caused by a typo `name !== field` was always true. Updated to `name !== field.name`. This fixes a bug were the very first column could not be unselected as the display column (because it was always the first column).

Addresses: 
- https://github.com/Budibase/budibase/issues/5302

## Screenshots
**BEFORE**
del_disp_before
*What a hassle!*

**AFTER**
del_disp

### CODE REVIEW

File: packages/builder/src/stores/backend/tables.js [hunk 1]
```diff
@@ -90,6 +90,23 @@ export function createTablesStore() {
     await fetch()
   }
 
+  const assignDisplayColumn = ({
```
Comment (shogunpurple) [inline packages/builder/src/stores/backend/tables.js:93]: Now that we have `vitest`, I think it would be good to see a unit test for this.

**Review by shogunpurple — COMMENTED** [review]
Functionality looks good to me once a unit test is added!

File: packages/builder/src/stores/backend/tables.js (hunk 1 above)
Comment (melohagan) [reply packages/builder/src/stores/backend/tables.js:93]: Unit tests added for the tables store

**Review by mike12345567 — COMMENTED** [review]
Brilliant set of tests, above and beyond what was required - great work!

Just one question about the way a random column is picked, I think we need some sort of type checking, so that we don't pick a display column which isn't valid.

File: packages/builder/src/stores/backend/tables.js [hunk 2]
```diff
@@ -90,6 +90,23 @@ export function createTablesStore() {
...
+    draft,
+    field,
+    originalName,
+  }) => {
+    if (primaryDisplay) {
+      draft.primaryDisplay = field.name
+    } else if (draft.primaryDisplay === originalName) {
+      const fields = Object.keys(draft.schema)
+      // pick another display column randomly if unselecting
+      draft.primaryDisplay = fields.filter(
```
[the remark this answers is not in the document]
Comment (melohagan) [reply packages/builder/src/stores/backend/tables.js:104]: Added a check to ignore attachment, json and link types.

File: scripts/releaseHelmChart.js [hunk 3]
```diff
@@ -1,28 +1,34 @@
-const yaml = require("js-yaml")
```
Comment (mike12345567) [inline scripts/releaseHelmChart.js:1]: I assume these changes are just linting?

**Review by mike12345567 — APPROVED** [review]
LGTM - nice test cases :)

File: scripts/releaseHelmChart.js (hunk 3 above)
Comment (melohagan) [reply scripts/releaseHelmChart.js:1]: Yes just linting

**Pull request merged**